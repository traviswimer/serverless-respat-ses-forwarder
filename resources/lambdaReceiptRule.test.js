const lambdaReceiptRule = require('./lambdaReceiptRule');

describe('lambdaReceiptRule()', () => {
	test('generates role', () => {
		expect(lambdaReceiptRule({
			prefix: "ThePrefix",
			pattern_name: "ThePatternName",
			lambda_function_name: "TheLambdaName",
			email_recipients: ["email@domain"],
			ses_ruleset_name: "TheRulesetName"
		})).toMatchSnapshot();
	});
});
