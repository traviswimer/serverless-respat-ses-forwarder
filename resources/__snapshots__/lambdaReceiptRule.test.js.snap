// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`lambdaReceiptRule() generates role 1`] = `
Object {
  "DependsOn": Array [
    "ThePatternNameEmailBucket",
    "ThePatternNameEmailBucketPolicy",
    "ThePatternNameBucketReceiptRule",
  ],
  "Properties": Object {
    "After": "ThePrefix-ThePatternName-s3-email-store",
    "Rule": Object {
      "Actions": Array [
        Object {
          "LambdaAction": Object {
            "FunctionArn": Object {
              "Fn::GetAtt": Array [
                "TheLambdaNameLambdaFunction",
                "Arn",
              ],
            },
            "InvocationType": "Event",
          },
        },
      ],
      "Enabled": true,
      "Name": "ThePrefix-ThePatternName-email-lambda-rule",
      "Recipients": Array [
        "email@domain",
      ],
      "ScanEnabled": true,
    },
    "RuleSetName": "TheRulesetName",
  },
  "Type": "AWS::SES::ReceiptRule",
}
`;
